Simplifying the documentation (#87)
Azure/Azure-Orbital-Analytics-Samples: deploy/README.md

# Overview of Infrastructure deployment and configuration

We are using bicep templates to deploy the infrastructure resources.
We will be using following azure services:
- Azure Batch Account
- Azure Synapse Analytics
- Azure Keyvault
- Azure Storage Account
- Azure Networking

> - Azure Log Analytics

Author: I thought of listing here so that user is aware about what bicep template is deploying.

# Prerequisites

The deployment script uses following tools, please follow the links provided to install the suggested tools on your computer using which you would execute the script.

- [git](https://github.com/git-guides/install-git) 
- [az cli](https://docs.microsoft.com/cli/azure/install-azure-cli)
- [jq](https://stedolan.github.io/jq/download/)
- The scripts are executed on bash shell, so if using a computer with windows based operating system, install a [WSL](https://docs.microsoft.com/windows/wsl/about) environment to execute the script.

> - Install bicep on az cli using command `az bicep install`.

Reviewer: ```suggestion
- Install bicep on az cli using command  `az bicep install` and ensure the bicep version is >=0.8.9.
```
and remove the next bullet point

- The bicep templates have been written to adhere to the syntax and rules for bicep version >= 0.8.9. Please check your bicep version using `az bicep version` if you run into bicep related errors.

> - The user performing the deployment of the bicep template and the associated scripts should have `Owner` role assigned at the subscription to which the resources are being deployed.

Reviewer: Move this line to a separate section after prereqs similar to [this](https://github.com/Azure/Azure-Orbital-STAC/tree/main/deploy#rbac-requirement)

> - This solution assumes no interference from Policies deployed to your tenant preventing resources from being deployed. 

Reviewer: we decided to move this line to troubleshooting.md and update as `TODO` so later we can add more details

- Get the repository to find the scripts. Clone the repository using following command.

```bash
git clone git@github.com:Azure/Azure-Orbital-Analytics-Samples.git
```

Alternatively, one can use Azure Cloud Bash to deploy this sample solution in their Azure subscription.

## Infrastructure Deployment

> Before executing the script one would need to login to azure using `az` cli and set the subscription in which they want to provision the resources.

Reviewer: ```suggestion
Before executing the script, please login to azure using `az` cli and set the subscription in which you want to provision the resources.
```

```bash
az login
az account set -s <subscription_id>
```

> To install, configure and package the infrastructure for dependencies one can execute the script [setup.sh](./setup.sh)

Reviewer: Can we update lines from 41 to 54 to something like below:

The following command will install, configure and generate the custom vision model package.

```bash
./deploy/setup.sh <environmentCode> <location>
```
- `environmentCode` used as a prefix for Azure resource names. It allows only alpha numeric(no special characters) and must be between 3 and 8 characters.
- `location` is a valid Azure region.

For eg.
```bash
./deploy/setup.sh aoi eastus
```

Script has been written to be executed with minimalistic input, it requires following input
- `environmentCode` which serves as the prefix for infrastructure services names. Allows only alpha numeric(no special characters) and must be between 3 and 8 characters.
- `location` which suggests which azure region infrastructure is deployed in.

```bash
./deploy/setup.sh <environmentCode> <location>
```

For eg.
```bash
./deploy/setup.sh aoi eastus
```

[setup.sh](./setup.sh) executes tasks in 3 steps
- installs the infrastructure using [install.sh](./install.sh) script.
- configures the infrastructure for setting up the dependecies using [configure.sh](./configure.sh) script.

> - packages the pipeline code to imported into synapse for standalone executions using [package.sh](./package.sh) script.

Reviewer: I'd just say the package script will generate the custom vision model package and not have reference to importing into Synapse.

> **Note**: After the script is completed without errors, please check the batch-account pool created is created succesfully.

Reviewer: ```suggestion
**Note**: After the setup script has run successfully, please check `<envCode>-orc-rg`, select the batch account service and ensure the batch pool has been created successfully.
```

> One can find the batch account named as `batchact<10-character-random-string>` in the resource-group `<environment-code>-orc-rg`.

Reviewer: ```suggestion 
You can find the batch account named as `batchact<10-character-random-string>` under the resource group `<environment-code>-orc-rg`.
```

- Click on `Pipelines` in the top menu.
- In the search box type `space` to find the pipeline named `Spaceborne Data Analysis Master Pipeline`.
- Click on pipeline name to select it, and press `Continue` button at the end of the page.

Follow the steps in [instructions.md](./gallery/instructions.md) to configure and execute the pipeline.

## Cleanup Script

We have a cleanup script to cleanup the resource groups and thus the resources provisioned using the `environmentCode`.
As discussed above the `environmentCode` is used as prefix to generate resource group names, so the cleanup-script deletes the resource groups with generated names.

Execute the cleanup script as follows:

```bash
./deploy/cleanup.sh <environmentCode>
```

For eg.
```bash
./deploy/cleanup.sh aoi
```

If one wants not to delete any specific resource group and thus resource they can use NO_DELETE_*_RESOURCE_GROUP environment variable, by setting it to true

```bash
NO_DELETE_DATA_RESOURCE_GROUP=true
NO_DELETE_MONITORING_RESOURCE_GROUP=true
NO_DELETE_NETWORKING_RESOURCE_GROUP=true
NO_DELETE_ORCHESTRATION_RESOURCE_GROUP=true
NO_DELETE_PIPELINE_RESOURCE_GROUP=true
./deploy/cleanup.sh <environmentCode>
```

> ## AI Model

Reviewer: Do we still need this section? @mandarinamdar @senthilkungumaraj

Reviewer 2: Its already covered in main README, we can remove this from this deploy readme section

Author: @mandarinamdar we decided to remove from the main README

Reviewer 2: In the main README, it is covered under GETTING started. I think that is repeating here and we are adding just last line for bring your own model. But bring your own model is again not covering that alone but covering additional content to use existing model in ACR/github.

Reviewer 2: Lets call out only this part of that section

**Follow the [document](https://github.com/sushilkm/Azure-Orbital-Analytics-Samples/blob/suskuma/doc-fixes/deploy/bring-your-own-ai-model.md) to understand and use custom vision model (Hyperlink) or bring different ai-model for processing with the pipeline.** 

Also lets rename  bring-your-own-ai-model.md to **using-ai-model-in-pipeline.md or something similar**
